test/extensions/formatter/req_without_query/req_without_query_test.cc

```diff
@@ -0,0 +1,104 @@
+#include "envoy/config/core/v3/substitution_format_string.pb.validate.h"
+
+#include "common/formatter/substitution_format_string.h"
+
+#include "test/mocks/server/factory_context.h"
+#include "test/mocks/stream_info/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Formatter {
+
+class ReqWithoutQueryTest : public ::testing::Test {
+public:
+  Http::TestRequestHeaderMapImpl request_headers_{
+      {":method", "GET"},
+      {":path", "/request/path?secret=parameter"},
+      {"x-envoy-original-path", "/original/path?secret=parameter"}};
+  Http::TestResponseHeaderMapImpl response_headers_;
+  Http::TestResponseTrailerMapImpl response_trailers_;
+  StreamInfo::MockStreamInfo stream_info_;
+  std::string body_;
+
+  envoy::config::core::v3::SubstitutionFormatString config_;
+  NiceMock<Server::Configuration::MockFactoryContext> context_;
+};
+
+TEST_F(ReqWithoutQueryTest, TestStripQueryString) {
+  const std::string yaml = R"EOF(
+  text_format_source:
+    inline_string: "%REQ_WITHOUT_QUERY(:PATH)%"
+  formatters:
+    - name: envoy.formatter.req_without_query
+      typed_config:
+        "@type": type.googleapis.com/google.protobuf.StringValue
```

Comment:
I did not find a way to avoid requiring user to add a placeholder config like empty `StringValue`.  Is that unavoidable, since the extension point for writing formatter commands `CommandParserFactory` inherits from `Config::TypedFactory` ([link](https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/include/envoy/formatter/substitution_formatter.h#L117)) instead of `UntypedFactory`?

Reply:
we need to add a config message, even if empty (e.g.: `message RequestWithoutQuery`)

Reply:
I added empty `ReqWithoutQuery` message and `#extension-category: envoy.formatter` to `ExtensionFilter` to get document generation working.